### PR TITLE
Add HSA names to API

### PR DESCRIPTION
**Pull request by smcclure17**
Adds names of HSAs to county endpoints, (or "Unknown HSA Name" when an HSA name is not available.) HSA names lined up/were available for all HSAs except those in Puerto Rico.

Tested by generating API files locally with:
```
./run.py data update --state CT
python ./pyseir/cli.py build-all --states=CT
./run.py api generate-api-v2 --state CT output -o output/api
```

County endpoint:
image

State endpoint:
image

I can generate a full snapshot for this, but I feel pretty confident that this should work as expected

### CODE REVIEW

File: api/can_api_v2_definition.py [hunk 1]
```diff
@@ -532,6 +532,10 @@ class RegionSummary(base_model.APIBaseModel):
         ...,
         description="3 digit Health Service Area identification code. For CBSA, state, and country regions hsa is omitted. For more on HSAs see: https://github.com/covid-projections/covid-data-model/blob/main/data/misc/README.md",
     )
+    hsaName: Optional[str] = pydantic.Field(
+        ...,
+        description="Name of Health Service Area. For CBSA, state, and country regions hsa is omitted. For more on HSAs see: https://github.com/covid-projections/covid-data-model/blob/main/data/misc/README.md",
```
Comment (mikelehen) [inline api/can_api_v2_definition.py:537]: "hsa is omitted" => "hsaName is omitted" ?

File: api/can_api_v2_definition.py [hunk 2]
```diff
@@ -642,6 +646,11 @@ class RegionTimeseriesRowWithHeader(base_model.APIBaseModel):
         description="3 digit Health Service Area identification code. For CBSA, state, and country regions hsa is omitted. For more on HSAs see: https://github.com/covid-projections/covid-data-model/blob/main/data/misc/README.md",
     )
 
+    hsaName: Optional[str] = pydantic.Field(
+        ...,
+        description="Name of Health Service Area. For CBSA, state, and country regions hsa is omitted. For more on HSAs see: https://github.com/covid-projections/covid-data-model/blob/main/data/misc/README.md",
```
Comment (mikelehen) [inline api/can_api_v2_definition.py:651]: ditto

File: data/misc/mapping_temp.csv [hunk 3]
```diff
@@ -0,0 +1,3221 @@
+FIPS,HSA,hsa_name
+1001,171,"Montgomery (Montgomery), AL - Autauga, AL"
```
Comment (mikelehen) [inline data/misc/mapping_temp.csv:2]: Is this `mapping_temp.csv` file meant to be checked in?

File: libs/datasets/sources/hsa_population.py [hunk 4]
```diff
@@ -40,6 +42,12 @@ def make_dataset(cls) -> timeseries.MultiRegionDataset:
         populations = FIPSPopulation.make_dataset().static
         county_index = populations.index.map(get_location_level) == AggregationLevel.COUNTY
         counties = populations[county_index].copy()  # copy to remove SettingWithCopy error
+        hsa_names: pd.DataFrame = (
+            pd.read_csv(HSA_LIST_PATH, dtype={CommonFields.HSA: str})
+            .loc[:, [CommonFields.HSA, CommonFields.HSA_NAME]]
+            .drop_duplicates()
+            .assign(hsa_name=lambda row: row[CommonFields.HSA_NAME].fillna("Unknown HSA Name"))
```
Comment (mikelehen) [inline libs/datasets/sources/hsa_population.py:49]: Can you make this just "Unknown"?

I'm imagining in the frontend we're going to have copy something like:
> Over the past 7 days, the {hsaName} Health Service Area had xyz staffed beds and xyz covid patients, resulting in xy% of staffed beds filled with covid patients.

**Review by mikelehen — APPROVED** [review]
Nice!  Looks good.

File: data/misc/mapping_temp.csv (hunk 3 above)
Comment (smcclure17) [reply data/misc/mapping_temp.csv]: nope it is not :( thanks

File: api/can_api_v2_definition.py (hunk 1 above)
Comment (smcclure17) [reply api/can_api_v2_definition.py]: ah, thanks for the catch--updated in 9aeffdab7a5a35dcab0e502b1f7f5cd9ccc57eb2

**Pull request merged**